sdk/src/azure/iot/az_iot_provisioning_client.c

```diff
@@ -276,6 +277,31 @@ AZ_INLINE az_result _az_iot_provisioning_client_parse_payload_error_code(
   return AZ_ERROR_ITEM_NOT_FOUND;
 }
 
+AZ_INLINE az_result
+_az_iot_provisioning_client_get_json_object_span(az_json_reader* jr, az_span* out_span)
+{
+  *out_span = jr->token.slice;
+
+  if (jr->token.kind == AZ_JSON_TOKEN_NULL)
+  {
+    *out_span = AZ_SPAN_EMPTY;
+    return AZ_OK;
+  }
+
+  if (jr->token.kind != AZ_JSON_TOKEN_BEGIN_OBJECT)
+  {
+    return AZ_ERROR_UNEXPECTED_CHAR;
+  }
```

Comment:
Code coverage for `_az_iot_provisioning_client_get_json_object_span` can be improved. Line coverage for this file is great at over 95%, but branch coverage can be better (around 70%).

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2104943&view=codecoverage-tab

![image](https://user-images.githubusercontent.com/6527137/211675764-1c42f879-d016-487f-b27a-6d33e5622349.png)

Reply:
What is the expected branch coverage for this repo? For example az_http_XXX, along with base64 parsing and json parser components has very low code and branch coverage. Are you guys addressing this???

Reply:
Thanks @ahsonkhan. We'll consider improving our code coverage in the future.

Reply:
The previous IoT changes/PRs from folks on the team had set a much higher bar for coverage, at close to 100% line coverage and 80%+ branch coverage, so I just raised my expectations accordingly for anything IoT related 😄 

Look at IoT Hub Client files, for example. They are pretty good!

> What is the expected branch coverage for this repo?
> If there are specific numbers, please enforce them through the CI system.
 
Agreed! **They are already here, enforced in CI**, repo-wide at 90%+ for line coverage and 70%+ for branch coverage. This file meets that target, but is at the border.

https://github.com/Azure/azure-sdk-for-c/blob/c551dfe9bd1e3b4152613066ec626d7c4901f421/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L369-L387

We are above that currently, in aggregate/repo-wide, but whenever I see some low hanging/useful test coverage improvement, I point it out to leave slack for places which are more challenging (and less fruitful) to test. We don't want each PR to incrementally trickle down towards that bar, and then a small change trips the CI failure for test coverage.

> For example az_http_XXX, along with base64 parsing and json parser components has very low code and branch coverage. Are you guys addressing this???

I wouldn't consider these "very low", in fact they are well above the bar:
- base64 has 97.7 and 81.7 respectively
- json reader has 97.5 and 91.5 respectively

Both base64 and json have pretty good overall coverage. Some areas can be improved, but most of the lack of branch coverage is around preconditions. Even then, we do have [specific issues to track that in the backlog](https://github.com/Azure/azure-sdk-for-c/issues?q=is%3Aopen+is%3Aissue+label%3Atest-enhancement).
For example:
https://github.com/Azure/azure-sdk-for-c/issues/1836
https://github.com/Azure/azure-sdk-for-c/issues/2310

That said, you correctly point out that the `az_http_policy_XXX` are the only ones that are lagging, but not enough attention has been put on those in the `main` branch for a while now, since we don't have any service being supported/shipping out of main that uses HTTP. For example, the entire `az_http_pipeline_policy_credential()` is uncovered. It also currently has no callers. Once there is a motivating need to revisit and modify those files (possibly with storage APIs being added), we should definitely up the coverage there!